Rename symbols to allow Apple App Store uploads

elevenfive opened the pull request:
### Description

- Apple's App Store validation rejects binaries that reference certain symbol names (like JxlDecoder*) as they conflict with private APIs.

- This patch renames all public libjxl symbols with an SDL_ prefix when building for Apple platforms to avoid these conflicts.

<img width="1442" height="620" alt="Screenshot 2025-11-08 at 6 04 48 PM" src="https://github.com/user-attachments/assets/13ddb808-ed61-47b1-b21d-91c9c774827d" />

sezero commented:
This patch is wrong, as far as I can see

(i) First of all, the patch should have been against 0.7.2-SDL branch which SDL_image uses

More importantly:

(ii) You are renaming libjxl exports within libjxl build, but SDL_image isn't aware of that and will fail linkage -- unless I'm missing something.

Besides: Will linking libjxl to SDL_image statically in iOS builds solve this issue properly?

/cc @slouken 
/cc @madebr

madebr commented:
Yes, this is not a correct approach. Jxl symbols are evidently not owned by Apple because of a broken check.

It looks like you are linking SDL3_image to libjxl at build time and not loading it dynamically.
Does building SDL_image with `-DSDLIMAGE_JXL_SHARED=ON` (or the equivalent in the SDL3_image xcode project) also trigger this check?

sezero commented:
BTW: Is SDL_image Apple-targeting builds linking statically to libjxl leaking libjxl symbols? AFAIR, we aren't providing SDL_image linkage with a symbols list in Apple-targeting builds. If such a thing is the case, and especially if the issue here is actually that, can we circumvent it by defining `JXL_STATIC_DEFINE` in vendored builds?

elevenfive committed:
Update CMakeLists.txt to fix Apple App Store issue

- Apple's App Store validation rejects binaries that reference certain symbol names (like JxlDecoder*) as they conflict with private APIs.

- This patch renames all public libjxl symbols with an SDL_ prefix when building for Apple platforms to avoid these conflicts.

elevenfive commented:
Thank you for looking and thinking with me here.  FWIW - I was attempting to simply use the `SDL_image` framework in the same manner as `SDL`.

**Historical background**

Starting on September 16, 2024 with iOS 18 and Xcode 16, Apple bundled libjxl (JPEG XL decoder library) into iOS as an internal/private API. When apps are submitted to the App Store, Apple's automated review process scans for symbols matching these private APIs and rejects apps that appear to use them - even if the app is using its own bundled version of libjxl rather than calling Apple's private APIs.

**What now?**

I was able to come up with a workaround in my project by building SDL_image and  renaming the symbols, which is working for my usecase but isn't long term sustainable.

I really appreciate you all taking a look so far.  I'm not sure exactly the right path to go from here.  Glad to experiment more with you all.  

This PR is definitely step 1 and if we went with a rename fix like this, the 2nd step would be to patch `SDL_image` to use the renamed symbols.

elevenfive commented:
Closing in favor of alternate changes in the works
